### ISSUE TITLE
[Question] What prepare_inputs_labels_for_multimodal in llava_arch.py does with labels?

### ISSUE THREAD

**Issue by EDDChang** [haotian-liu/LLaVA#893]
### Question

In my understanding, forward function may return loss if given labels., which is probability that llm generate the labels.
However, in the class LlavaLlamaForCausalLM, you override forward function with some preprocess of inputs by prepare_inputs_labels_for_multimodal and also process the labels.
I would like to know how the function "prepare_inputs_labels_for_multimodal" do to labels? 
In more specific, in line:144 and 145
input_ids = [cur_input_ids[cur_attention_mask] for cur_input_ids, cur_attention_mask in zip(input_ids, attention_mask)]
labels = [cur_labels[cur_attention_mask] for cur_labels, cur_attention_mask in zip(labels, attention_mask)]
why assuming input_ids and labels has same size?

**Comment by EDDChang** [comment]
problem solved.

**Comment by hhy-huang** [comment]
> problem solved.

Hi, thank you for your question. Now I have the same issue, could you explain how you figure it out?
